edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/Exceptions/MqttException.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter.Exceptions
+{
+    public class MqttException : Exception
+    {
+        public MqttException(bool isTemporary = false, string message = null, Exception cause = null) : base(message, cause)
+        {
+            IsTemporary = isTemporary;
+        }
+
+        public bool IsTemporary { get; }
+    }
```

Comment:
We don't have retry logic now since any operation(reconnect and pub/sub) failure in connector is not handled besides logging. Ideally we should triage the failure. The flag is defined for future improvement.